language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/TypeGuardCodeAction.java

```diff
@@ -175,10 +172,26 @@ public String getName() {
         SemanticModel semanticModel = context.currentSemanticModel().orElseThrow();
         Document srcFile = context.currentDocument().orElseThrow();
         Optional<Symbol> symbol = semanticModel.symbol(srcFile,
-                                                       assignmentStmtNode.varRef().lineRange().startLine());
+                assignmentStmtNode.varRef().lineRange().startLine());
         if (symbol.isEmpty() || symbol.get().kind() != SymbolKind.VARIABLE) {
             return Optional.empty();
         }
         return Optional.of((VariableSymbol) symbol.get());
     }
+
+    private boolean isInValidContext(NonTerminalNode matchedNode, CodeActionContext context) {
+
+        Optional<SyntaxTree> syntaxTree = context.currentSyntaxTree();
+        if (syntaxTree.isEmpty()) {
+            return false;
+        }
+        Node node = null;
+        if (matchedNode.kind() == SyntaxKind.ASSIGNMENT_STATEMENT) {
+            node = ((AssignmentStatementNode) matchedNode).varRef();
+        } else if (matchedNode.kind() == SyntaxKind.LOCAL_VAR_DECL) {
+            node = ((VariableDeclarationNode) matchedNode).typedBindingPattern().bindingPattern();
+        }
+        return node == null ? false : CommonUtil.isWithInRange(node, context.cursorPositionInTree());
```

Comment:
`node != null && CommonUtil.isWithInRange(node, context.cursorPositionInTree())`
What if we use this?